lib/rules/no-inner-declarations.js

```diff
@@ -66,6 +76,10 @@ module.exports = {
 
     create(context) {
 
+        const sourceCode = context.sourceCode;
+        const ecmaVersion = context.languageOptions.ecmaVersion ?? 6;
```

Comment:
```suggestion
        const ecmaVersion = context.languageOptions.ecmaVersion;
```

`context.languageOptions.ecmaVersion` is always a number: 3, 5, 2015, 2016...